GetStream/stream-chat-swift
# [CIS-1183] Add view to show all reactions of a message

### 🔗 Issue Link
CIS-1183

### 🎯 Goal
Add a view to show all reactions of a message and the author of that reaction, when tapping the reactions bubble view of a message.

### 🛠 Implementation
- Adds a new `MessageController.reactions` property that gets all the reactions of the message.
- Adds a new ` MessageController.loadReactions()` function to get reactions from the Backend given a  limit and offset
- Adds a new `MessageController.loadNextReactions()` to paginate the loading of all the reactions. When the user is scrolling all the reactions, it fetches more reactions until all reactions have been loaded.
- Adds a new UI Component `ChatMessageReactionAuthorsVC` to render all the reactions of the message along with its authors.
- Changes the `ChatMessagePopupVC` to support rendering the reaction authors.
- Adds new reaction documentation and also restructures the reactions docs.

### 🧪 Testing
1. Open a Channel
2. Tap on the reactions view of a message
3. A popup view controller should be presented with all the reactions.

### 🎨 Changes
**Video displaying all the edge cases of the `ChatMessageReactionAuthorsVC`:**

https://user-images.githubusercontent.com/12814114/138963810-9f1808c4-4e67-41f5-afcd-7c11720c26f8.mp4

### ☑️ Checklist

- [x] New code is covered by UI Tests
